From bbbe55e43d0b24f48fa764e7de12b4e11672b598 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 27 Nov 2015 15:57:02 +0530
Subject: [PATCH 04/12] [USBC]: Disable drp toggle thread

If the typec phy support drp toggle dont run the thread.

Change-Id: Ic1516a9496e4d943e4f67f8795a4295f015346d0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-8683
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/441811
---
 drivers/usb/typec/usb_typec_detect.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index fe8d38f..14999b3 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -927,11 +927,12 @@ int typec_bind_detect(struct typec_phy *phy)
 	INIT_WORK(&detect->drp_work, detect_drp_work);
 	INIT_WORK(&detect->ufp_work, detect_ufp_work);
 
-	if (!phy->support_drp_toggle)
+	if (!phy->support_drp_toggle) {
 		setup_timer(&detect->drp_timer, detect_drp_timer,
 				(unsigned long)detect);
-
-	detect->detect_kthread = kthread_run(detect_kthread, detect, "detect");
+		detect->detect_kthread = kthread_run(detect_kthread,
+							detect, "detect");
+	}
 	detect->state = DETECT_STATE_UNATTACHED_DRP;
 
 	detect->otg = usb_get_phy(USB_PHY_TYPE_USB2);
-- 
1.9.1

